### PR TITLE
Fix region error braintree

### PR DESCRIPTION
**Pull request by larsroettig**
## Description
When shipping information is set to a UK address, a region error is displayed within the Braintree Payment section, after clicking "Review Order"

Variable "$region" got invalid value null; Expected non-nullable type String! not to be null.

 https://develop.pwa-venia.com/
 https://venia.magento.com/
 

Steps to reproduce

Add item to cart (develop.pwa-venia.com)
Go to checkout
Enter UK address for Shipping Information
Set any Shipping method
Enter credit card (Braintree) info, and check Billing address same as shipping
Click "Review Order"
 Form error message displays
 Able to review, and place order

## Related Issue
Closes PWA-1823

## Acceptance
- Able to complete checkout with UK, France, and US addresses.
- Check with different shipping and billing addresses
- Update MFTF test to account for UK address

#### Test scenario(s) for direct fix/feature
Example
-   Verify user is able to apply filters on category page
-   Verify user is able to apply filters on search page

#### Test scenario(s) for any existing impacted features/areas
Example
-   Verify user is able to sort data after applying filters on category page
-   Verify user is able to sort data after applying filters on search page

#### Test scenario(s) for any Magento Backend Supported Configurations
Example
- Update default Sort value in backend and repeat above scenarios

#### Is Browser/Device testing needed?
Example
- Yes, browser testing is needed as X UI component may be impacted on <browser>
- Yes, device testing is needed as X UI component may be impacted on <mobile|desktop|etc>

#### Any ad-hoc/edge case scenarios that need to be considered?
Example
- Apply all filters to get 0 results and then remove filters to see respective products 

## Checklist

-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.

### CODE REVIEW

File: packages/peregrine/lib/talons/CheckoutPage/ShippingInformation/AddressForm/useGuestForm.js [hunk 1]
```diff
@@ -45,7 +45,8 @@ export const useGuestForm = props => {
                         address: {
                             ...address,
                             // region_id is used for field select and region is used for field input
```
Comment (anthoula) [inline packages/peregrine/lib/talons/CheckoutPage/ShippingInformation/AddressForm/useGuestForm.js:47]: I wonder if this comment is still relevant, as it looks like we are removing `region.region`. Perhaps update comment to reflect what is now happening

**Review by anthoula — APPROVED** [review]

**Comment by supernova-at** [comment]
## QA

📝  In Progress

- [ ] MFTF
- [ ] Cypress
- [ ] Test Plan
- [ ] Lighthouse
- [ ] Alternate Browsers
